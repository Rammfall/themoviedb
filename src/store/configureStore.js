import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
