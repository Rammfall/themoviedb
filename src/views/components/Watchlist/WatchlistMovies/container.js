import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import MoviesList from 'Views/components/MoviesList'
import { getWatchlistMoviesSelector } from 'Store/concepts/movies/selectors'

class WatchlistMovies extends Component {
  render() {
    const { movies: { total, movies } } = this.props

    return (
      <MoviesList
        movies={movies}
        total={total}
      />
    )
  }
}

WatchlistMovies.propTypes = {
  movies: PropTypes.shape({
    total: PropTypes.number.isRequired,
    movies: PropTypes.arrayOf().isRequired
  }).isRequired
}

const mapStateToProps = (state) => ({
  movies: getWatchlistMoviesSelector(state)
})

export { WatchlistMovies as WatchlistMoviesContainer }
export default connect(mapStateToProps)(WatchlistMovies)
