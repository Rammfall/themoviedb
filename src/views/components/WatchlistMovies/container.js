import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { compose } from 'ramda'
import { injectIntl } from 'react-intl'
import { Modal } from 'antd'
import { DeleteOutlined } from '@ant-design/icons'

import MoviesList from 'Views/components/MoviesList'
import { getWatchlistMoviesSelector } from 'Store/concepts/watchlist/selectors'
import { toggleWatchlistMovie } from 'Store/concepts/watchlist/actions'
import onConfirm from 'Utils/components/modalsHandlers/onConfirm'

class WatchlistMovies extends Component {
  handleRemoveFromList = (id) => () => {
    const { intl: { formatMessage }, toggleWatchlist } = this.props

    Modal.confirm({
      title: formatMessage({ id: 'watchlist.remove.title' }),
      onOk: onConfirm(id, toggleWatchlist)
    })
  }

  actions = (id) => [
    <DeleteOutlined
      key="delete"
      onClick={this.handleRemoveFromList(id)}
    />
  ]

  render() {
    const { total, movies } = this.props

    return (
      <MoviesList
        movies={movies}
        total={total}
        actions={this.actions}
      />
    )
  }
}

WatchlistMovies.propTypes = {
  total: PropTypes.number.isRequired,
  movies: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    original_title: PropTypes.string,
    overview: PropTypes.string,
    poster_path: PropTypes.string
  })).isRequired,
  intl: PropTypes.shape({
    formatMessage: PropTypes.func.isRequired
  }).isRequired,
  toggleWatchlist: PropTypes.func.isRequired
}

const mapStateToProps = (state) => ({
  movies: getWatchlistMoviesSelector(state).movies,
  total: getWatchlistMoviesSelector(state).total
})

const dispatchToProps = {
  toggleWatchlist: toggleWatchlistMovie
}

export { WatchlistMovies as WatchlistMoviesContainer }
export default compose(
  injectIntl,
  connect(mapStateToProps, dispatchToProps)
)(WatchlistMovies)
