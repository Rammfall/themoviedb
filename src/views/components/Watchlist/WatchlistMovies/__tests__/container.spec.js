import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import diveTo from 'TestUtils/diveToEnzyme'

import WatchlistMovies, { WatchlistMoviesContainer } from '../container'

jest.mock('antd', () => ({
  Modal: {
    confirm: jest.fn()
  }
}))

jest.mock('Store/concepts/movies/actions', () => ({
  toggleWatchlistMovie: jest.fn()
}))

jest.mock('Store/concepts/movies/selectors', () => ({
  getWatchlistMoviesSelector: () => ({
    movies: [{
      id: 1,
      original_title: 'test',
      overview: 'test',
      poster_path: 'test'
    }],
    total: 1
  })
}))

jest.mock('react-intl', () => ({
  get injectIntl() {
    return component => component
  }
}))

describe('WatchlistMovies', () => {
  const store = configureStore()()
  const props = {
    intl: {
      formatMessage: jest.fn()
    }
  }

  it('matches snapshot', () => {
    const wrapper = shallow(
      <WatchlistMovies
        store={store}
        {...props}
      />
    )
    const container = diveTo(wrapper, WatchlistMoviesContainer)

    expect(container).toMatchSnapshot()
  })

  it('handleRemoveFromList', () => {
    const wrapper = shallow(
      <WatchlistMovies
        store={store}
        {...props}
      />
    )
    const container = diveTo(wrapper, WatchlistMoviesContainer)
    const spy = jest.spyOn(container.instance(), 'handleRemoveFromList')
    container.instance().handleRemoveFromList(3)()

    expect(spy).toHaveBeenCalledWith(3)
  })

  it('actions', () => {
    const wrapper = shallow(
      <WatchlistMovies
        store={store}
        {...props}
      />
    )
    const container = diveTo(wrapper, WatchlistMoviesContainer)
    const spy = jest.spyOn(container.instance(), 'actions')
    container.instance().actions(3)

    expect(spy).toHaveBeenCalledWith(3)
  })
})
