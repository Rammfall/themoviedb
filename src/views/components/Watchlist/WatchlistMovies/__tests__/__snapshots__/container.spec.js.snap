// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchlistMovies matches snapshot 1`] = `
<MoviesList
  actions={[Function]}
  movies={
    Array [
      Object {
        "id": 1,
        "original_title": "test",
        "overview": "test",
        "poster_path": "test",
      },
    ]
  }
  total={1}
/>
`;
